source/extensions/transport_sockets/tls/cert_validator/default_validator.cc

```diff
@@ -530,7 +532,15 @@ void DefaultCertValidator::addClientValidationContext(SSL_CTX* ctx, bool require
   }
   // Set the verify_depth
   if (config_->maxVerifyDepth().has_value()) {
-    SSL_CTX_set_verify_depth(ctx, config_->maxVerifyDepth().value());
+    uint32_t max_verify_depth = std::min(config_->maxVerifyDepth().value(), uint32_t{INT_MAX});
```

Comment:
nit: const

Reply:
Already mentioned in another comment thread. The value is not const. See the immediately following line.